firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -41,22 +41,42 @@ void moveDiagonal(int dirX, int dirY, int spacesX, int spacesY)
 
   // similary subject to change as above function dictation, same number as above though
   float numStepsX = spacesX * stepsPerSpace;
-  float numStepsX = spacesY * stepsPerSpace;
+  float numStepsY = spacesY * stepsPerSpace;
 
   digitalWrite(MOTOR_SLEEP, HIGH);
   digitalWrite(MOTOR_RESET, HIGH);
   digitalWrite(MOTOR_ENABLE, LOW);
 
   digitalWrite(xMotor[1], dirX);
-  digitalWrite(yMotor[1], dirX);
-
-  for(i = 0; i < numSteps; i++)
+  digitalWrite(yMotor[1], dirY);
+  
+  if (numStepsX = numStepsY) 
   {
+    for(i = 0; i < numStepsX; i++)
+    {
     digitalWrite(xMotor[0], LOW);
     digitalWrite(yMotor[0], LOW);
     delay(1);
     digitalWrite(xMotor[0], HIGH);
     digitalWrite(yMotor[0], HIGH);
+    }
+  } 
+
+  else 
+  {
+    float dist = sqrt(numStepsX * numStepsX + numStepsY * numStepsY);
```

Comment:
I see what you're doing here, but I don't think we need to do this. We just need to make the motors move an x and y distance.

Reply:
Additionally, I talked to Nashir and we agreed that we don't need to worry about all slopes, just 1, 1/2, and 2. That being said, you want to set the motor scale's and drive them the same number of steps. With the scale different, one will do full steps and the other 1/2 steps. So the slope will be maintained. Or the other way around and that's a slope of 2.

Reply:
To do this, use if statements to check which is bigger and the slope between them, which is just y/x

Reply:
Sorry for the noob question, but do we need to explicitly consider negative slopes as well?